### DISCUSSION TITLE
Set decimal places independently for displayed numbers and charts

### DISCUSSION THREAD

**Discussion by Brezentrager** [poblabs/weewx-belchertown#870] [Ideas]
If I understand it correctly, you can currently only specify the number of desired decimal places once for each `unit_group`. This is then used both for the display of numbers on the main page and for the charts.

For example, since I want to display the temperature with only one decimal place, it is also used for the charts with only one decimal place. In the case of the charts, this leads to them looking very step-like.

Is there a possibility:
a) To define the number of decimal places separately for the display of numbers on the page and for use in charts?
b) Or to use one more decimal place by default for the charts than for the display / rounding of numbers?

**Comment by roe-dl** [comment]
example from `graphs.conf`:
```
    [[humidity]]
        title = "Luftfeuchtigkeit"
        [[[outHumidity]]]
            name = "relativ"
        [[[outHumAbs]]]
            name = "absolut"
            yAxis = 1
            unit = gram_per_meter_cubed
            [[[[numberFormat]]]]
                decimals = 1
```

**Reply by Brezentrager** [reply]
I've tried that. But in the charts, the values only change from 3.4 (degrees) to 3.40 (not to something like 3.42)

**Reply by roe-dl** [reply]
What is the accuracy of the data to be displayed? Do they have 2 decimals? 

You can use the `sqlite3` tool to see what is in the database.

Example:
```
$ sqlite3 /var/lib/weewx/weewx.sdb 
SQLite version 3.31.1 2020-01-27 19:55:54
Enter ".help" for usage hints.
sqlite> select distinct windDir from archive;
```

**Reply by Brezentrager** [reply]
Yes, there is plenty of accuracy available.

When changing the string formatting in `weewx.conf` to `%.2f` for the respective `unit`, values with 2 decimals (e.g. `3.42`) are displayed. With the `%.1f` setting, a value of `3.40` gets plotted in the chart.

**Reply by roe-dl** [reply]
Actually I don't know.
